From: Simon Vogl <simon.vogl@gmx.net>
Date: Tue,  7 Dec 2021 21:41:49 UTC
Subject: [PATCH] Fix broken python detection of QMPlay2 when it tries to launch youtube-dl

This patch is required to prevent QMPlay2 from always preferring python2 over python3 which can lead to crashes in certain cases.
It also adds openSUSE branding to the about screen.

--- a/src/qmplay2/YouTubeDL.cpp
+++ b/src/qmplay2/YouTubeDL.cpp
@@ -467,29 +467,9 @@
     QString program = m_ytDlPath;

 #ifndef Q_OS_WIN
-    QFile ytDlFile(program);
-    if (ytDlFile.open(QFile::ReadOnly))
-    {
-        const auto shebang = ytDlFile.readLine(99).trimmed();
-        const int idx = shebang.lastIndexOf("python");
-        if (shebang.startsWith("#!") && idx > -1)
-        {
-            const auto pythonCmd = shebang.mid(idx);
-            if (QStandardPaths::findExecutable(pythonCmd).endsWith(pythonCmd))
-            {
-                args.prepend(program);
-                program = pythonCmd;
-            }
-#ifdef Q_OS_MACOS
-            else if (QFileInfo(QString("/usr/local/bin/" + pythonCmd)).isExecutable())
-            {
-                args.prepend(program);
-                program = "/usr/local/bin/" + pythonCmd;
-            }
-#endif
-        }
-        ytDlFile.close();
-    }
+    //Enforce execution command as "python3", disable automatic detection
+    args.prepend(program);
+    program = "python3";
 #endif

     m_process.start(program, args);

--- a/src/gui/AboutWidget.cpp
+++ b/src/gui/AboutWidget.cpp
@@ -50,6 +50,8 @@
 #endif
     labelText += ")";
     labelText += "<br/><br/><a href='https://github.com/zaps166/QMPlay2/releases/latest'>" + tr("GitHub download page") + "</a>";
+    labelText += "<br/><b>This release was supplied by SUSE/openSUSE</b>";
+    labelText += "<br/><b>and can be updated via the package manager.</b>";
     QLabel *label = new QLabel(labelText);
 
     QLabel *iconL = new QLabel;
